### ISSUE TITLE
require-atomic-updates false positive

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#11194]
**Tell us about your environment**

* **ESLint Version:** 5.10.0
* **Node Version:** 11.1.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**
**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Online demo: https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgcmVxdWlyZS1hdG9taWMtdXBkYXRlczogZXJyb3IgKi9cclxuXHJcbmFzeW5jIGZ1bmN0aW9uIGYoKSB7XHJcbiAgICBsZXQgcmVjb3Jkc1xyXG4gICAgXHJcbiAgICByZWNvcmRzID0gYXdhaXQgZGIucmVjb3Jkcy5nZXQoKVxyXG5cclxuICAgIGcoKCkgPT4geyByZWNvcmRzIH0pXHJcbn1cclxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19

```js
/*eslint require-atomic-updates: error */

async function f() {
    let records
    
    records = await db.records.get() //→ Possible race condition: `records` might be reassigned based on an outdated value of `records`. (require-atomic-updates)

    g(() => { records })
}
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

`require-atomic-updates` warns the line `records = await db.records.get()`.

If I change the `db.records.get` to `db.recordsDash.get` or something like, the warning is gone, so I guess that the rule uses the property name as a variable name.

**Are you willing to submit a pull request to fix this bug?**

I don't have enough time to investigate this for now. I'm sorry.
